src/client/java/de/rettichlp/pkutils/PKUtilsClient.java

```diff
@@ -9,36 +11,43 @@
 import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
 import net.minecraft.client.network.ClientPlayNetworkHandler;
 import net.minecraft.client.network.ClientPlayerEntity;
+import org.atteo.classindex.ClassIndex;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class PKUtilsClient implements ClientModInitializer {
 
     public static final Storage storage = new Storage();
+    private static final List<PKUtilsBase> listeners = new ArrayList<>();
 
     public static ClientPlayerEntity player;
     public static ClientPlayNetworkHandler networkHandler;
 
+    public static ActivityService activityService;
     public static FactionService factionService;
     public static SyncService syncService;
 
     private final Registry registry = new Registry();
 
     @Override
     public void onInitializeClient() {
-        // This entrypoint is suitable for setting up client-specific logic, such as rendering.
-
+        activityService = new ActivityService();
         factionService = new FactionService();
         syncService = new SyncService();
 
         ClientPlayConnectionEvents.JOIN.register((handler, sender, minecraftClient) -> minecraftClient.execute(() -> {
-            assert minecraftClient.player != null; // cannot be null at this point
+            assert minecraftClient.player != null;
             player = minecraftClient.player;
             networkHandler = minecraftClient.player.networkHandler;
 
             this.registry.registerListeners();
         }));
 
-        ClientCommandRegistrationCallback.EVENT.register((dispatcher, registryAccess) -> {
-            this.registry.registerCommands(dispatcher);
-        });
+        ClientCommandRegistrationCallback.EVENT.register((dispatcher, registryAccess) -> this.registry.registerCommands(dispatcher));
+    }
+
+    public static List<PKUtilsBase> getListeners() {
```

Comment:
Wozu wird das benötigt? Theoretisch gibt es in der Registry eine Liste aller Listeners